nodes/analyzer/proportional.py

```diff
@@ -105,19 +105,24 @@ def process(self):
 
             # build KDTree
             base = [v for v, mask in zip(vertices, masks) if mask]
-            tree = kdtree.KDTree(len(base))
-            for i, v in enumerate(base):
-                tree.insert(v, i)
-            tree.balance()
-
-            coeffs = []
-            for vertex, mask in zip(vertices, masks):
-                if mask:
-                    coef = 1.0
-                else:
-                    _, _, rho = tree.find(vertex)
-                    coef = self.falloff(radius, rho)
-                coeffs.append(coef)
+            
+            if len(base):
+
+                tree = kdtree.KDTree(len(base))
+                for i, v in enumerate(base):
+                    tree.insert(v, i)
+                tree.balance()
+
+                coeffs = []
+                for vertex, mask in zip(vertices, masks):
+                    if mask:
+                        coef = 1.0
+                    else:
+                        _, _, rho = tree.find(vertex)
+                        coef = self.falloff(radius, rho)
+                    coeffs.append(coef)
+
+            else: coeffs = [0 for _ in masks]
```

Comment:
make it 
```python
    else:
        coeffs = [0.0 for _ in masks]
```
we keep the else and the code it executes on different lines (per convention)

for consistency i'd user `0.0` as the coefficient